packages/discord-request/src/request.ts

```diff
@@ -0,0 +1,79 @@
+import type { ErrorBody } from "discord-error";
+import { DiscordError, isDiscordError } from "discord-error";
+import type { Client } from "./client.js";
+import { RateLimitError } from "./errors/rate-limit-error.js";
+import { RequestError } from "./errors/request-error.js";
+import { parse } from "./response.js";
+
+export async function request(
+	client: Client,
+	path: string,
+	init: RequestInit,
+	abortSignal: AbortSignal | null | undefined
+) {
+	// Setup the timeout signal
+	const signal = new AbortController();
+	const abortRequest = () => {
+		signal.abort();
+	};
```

Comment:
We could (and should), but I'm waffling on adding a callback for this later (onTimeout or something) - gonna leave it in for now because every time I see it it reminds me to think about it